File: app/src/main/java/com/kkndesasendang/sendangsmartlearning/ui/learn/selectMaterial/SelectMaterialFragment.java
package com.kkndesasendang.sendangsmartlearning.ui.learn.selectMaterial;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.animation.AnimationUtils;
import android.view.animation.LayoutAnimationController;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.kkndesasendang.sendangsmartlearning.R;
import com.kkndesasendang.sendangsmartlearning.model.AudioVisualContentModel;
import com.kkndesasendang.sendangsmartlearning.model.AudioVisualMaterialModel;
import com.kkndesasendang.sendangsmartlearning.model.MaterialModel;

import java.util.ArrayList;

public class SelectMaterialFragment extends Fragment implements MaterialListAdapter.OnItemClickCallback {
    private RecyclerView mRVMaterialList;
    private TextView mTVEmptyPrompt;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_select_material, container, false);
        mRVMaterialList = view.findViewById(R.id.selectMaterialList);
        mTVEmptyPrompt = view.findViewById(R.id.selectMaterialEmptyListPrompt);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        SelectMaterialViewModel mSelectMaterialViewModel = new ViewModelProvider(this).get(SelectMaterialViewModel.class);
        mSelectMaterialViewModel.setMaterialList();

        LayoutAnimationController animController = AnimationUtils.loadLayoutAnimation(requireContext(), R.anim.layout_animation_fall_down);

        mRVMaterialList.setLayoutManager(new LinearLayoutManager(requireContext()));
        mRVMaterialList.setHasFixedSize(true);
        mRVMaterialList.setLayoutAnimation(animController);
        final MaterialListAdapter adapter = new MaterialListAdapter();
        adapter.setOnItemClickCallback(this);
        mRVMaterialList.setAdapter(adapter);

        mSelectMaterialViewModel.getMaterialList().observe(getViewLifecycleOwner(), new Observer<ArrayList<MaterialModel>>() {
            @Override
            public void onChanged(ArrayList<MaterialModel> materialModels) {
                adapter.updateDataset(materialModels);
            }
        });
    }

    @Override
    public void onItemClick(MaterialModel data, View view) {
        if (data.getClass() == AudioVisualMaterialModel.class) {
            AudioVisualContentModel avContents[] = ((AudioVisualMaterialModel) data).getContents().toArray(new AudioVisualContentModel[0]);
            Navigation.findNavController(view).navigate(SelectMaterialFragmentDirections.actionNavSelectMaterialToNavAudioVisual(avContents, data.getMaterialName()));
        }
    }
}

File: app/src/main/java/com/kkndesasendang/sendangsmartlearning/ui/fragments/SelectMaterialFragment.kt
package com.kkndesasendang.sendangsmartlearning.ui.fragments

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.fragment.findNavController
import androidx.navigation.fragment.navArgs
import androidx.recyclerview.widget.LinearLayoutManager
import com.kkndesasendang.sendangsmartlearning.R
import com.kkndesasendang.sendangsmartlearning.model.GamaselaMaterialModel
import com.kkndesasendang.sendangsmartlearning.ui.rvAdapters.MaterialListAdapter
import kotlinx.android.synthetic.main.fragment_select_material.*

class SelectMaterialFragment : Fragment(), MaterialListAdapter.OnItemClickCallback {
    val args: SelectMaterialFragmentArgs by navArgs()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_select_material, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val materials = args.materials
        val topicName = args.topicName
        val rvAdapter = MaterialListAdapter()

        selectMaterialTitle.text = topicName
        selectMaterialList.layoutManager = LinearLayoutManager(requireContext())
        selectMaterialList.adapter = rvAdapter
        rvAdapter.setOnItemClickCallback(this)
        rvAdapter.updateDataset(materials)
    }

    override fun onItemClick(topic: GamaselaMaterialModel) {
        val action = SelectMaterialFragmentDirections.actionNavSelectSubMaterialToNavAudioVisual(topic)
        findNavController().navigate(action)
    }
}